From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 26 Oct 2021 22:56:00 +0200
Subject: [PATCH] Fix broken tests when running without gtest

    Error encountered: hooks libraries failed to validate - library or libraries in error are:
    src/bin/dhcp4/tests/.libs/libco1.so (src/bin/dhcp4/tests/test_config.json:13:5)

--- a/src/bin/dhcp4/tests/Makefile.am
+++ b/src/bin/dhcp4/tests/Makefile.am
@@ -20,11 +20,6 @@
 # Don't install tests.
 noinst_SCRIPTS = $(SHTESTS)
 
-if HAVE_GTEST
-
-# C++ tests
-PROGRAM_TESTS = dhcp4_unittests
-
 AM_CPPFLAGS = -I$(top_srcdir)/src/lib -I$(top_builddir)/src/lib
 AM_CPPFLAGS += -I$(top_srcdir)/src -I$(top_builddir)/src
 AM_CPPFLAGS += -I$(top_srcdir)/src/bin -I$(top_builddir)/src/bin
@@ -72,6 +67,14 @@
 libco3_la_CPPFLAGS = $(AM_CPPFLAGS)
 libco3_la_LDFLAGS = -avoid-version -export-dynamic -module -rpath /nowhere
 
+# Don't install test libraries.
+noinst_LTLIBRARIES = libco1.la libco2.la libco3.la
+
+if HAVE_GTEST
+
+# C++ tests
+PROGRAM_TESTS = dhcp4_unittests
+
 dhcp4_unittests_SOURCES  = d2_unittest.h d2_unittest.cc
 dhcp4_unittests_SOURCES += dhcp4_unittests.cc
 dhcp4_unittests_SOURCES += dhcp4_srv_unittest.cc
@@ -168,9 +171,6 @@
 DISTCLEANFILES += marker_file.h
 DISTCLEANFILES += test_data_files_config.h
 DISTCLEANFILES += test_libraries.h
-
-# Don't install test libraries.
-noinst_LTLIBRARIES = libco1.la libco2.la libco3.la
 
 # Don't install C++ tests.
 noinst_PROGRAMS = $(PROGRAM_TESTS)
--- a/src/bin/dhcp6/tests/Makefile.am
+++ b/src/bin/dhcp6/tests/Makefile.am
@@ -20,11 +20,6 @@
 # Don't install shell tests.
 noinst_SCRIPTS = $(SHTESTS)
 
-if HAVE_GTEST
-
-# C++ tests
-PROGRAM_TESTS = dhcp6_unittests
-
 AM_CPPFLAGS = -I$(top_srcdir)/src/lib -I$(top_builddir)/src/lib
 AM_CPPFLAGS += -I$(top_srcdir)/src/bin -I$(top_builddir)/src/bin
 AM_CPPFLAGS += -I$(top_srcdir)/src -I$(top_builddir)/src
@@ -73,6 +68,14 @@
 libco3_la_CPPFLAGS = $(AM_CPPFLAGS)
 libco3_la_LDFLAGS = -avoid-version -export-dynamic -module -rpath /nowhere
 
+# Don't install test libraries.
+noinst_LTLIBRARIES = libco1.la libco2.la libco3.la
+
+if HAVE_GTEST
+
+# C++ tests
+PROGRAM_TESTS = dhcp6_unittests
+
 # This list is ordered alphabetically. When adding new files, please maintain
 # this order.
 dhcp6_unittests_SOURCES  = classify_unittests.cc
@@ -168,9 +171,6 @@
 DISTCLEANFILES += marker_file.h
 DISTCLEANFILES += test_data_files_config.h
 DISTCLEANFILES += test_libraries.h
-
-# Don't install test libraries.
-noinst_LTLIBRARIES = libco1.la libco2.la libco3.la
 
 # Don't install C++ tests.
 noinst_PROGRAMS = $(PROGRAM_TESTS)
--- a/src/lib/dhcpsrv/tests/Makefile.am
+++ b/src/lib/dhcpsrv/tests/Makefile.am
@@ -20,7 +20,6 @@
         $(LIBTOOL) --mode=execute $(VALGRIND_COMMAND)
 
 TESTS =
-if HAVE_GTEST
 # Build shared libraries for testing. The libtool way to create a shared library
 # is to specify "-avoid-version -export-dynamic -module" in the library LDFLAGS
 # (see http://www.gnu.org/software/libtool/manual/html_node/Link-mode.html).
@@ -53,6 +52,8 @@
 libco3_la_CXXFLAGS = $(AM_CXXFLAGS)
 libco3_la_CPPFLAGS = $(AM_CPPFLAGS)
 libco3_la_LDFLAGS = -avoid-version -export-dynamic -module -rpath /nowhere
+
+if HAVE_GTEST
 
 TESTS += libdhcpsrv_unittests
 
